From 4f17b3fe5066440796d12c2119d66b05c56fe71a Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Thu, 27 Oct 2016 16:06:27 +0200
Subject: [PATCH 124/127] sfdisk: remove useless CDROM detection for -s

* close device on error
* don't use size (INTMAX) to detect non-mounted cdrom (just depend on
  blkdev_get_sectors() like we use use for new sfdisk implemenation

Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1319642
Signed-off-by: Karel Zak <kzak@redhat.com>
---
 fdisk/sfdisk.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/fdisk/sfdisk.c b/fdisk/sfdisk.c
index 061a04c..4fcbca5 100644
--- a/fdisk/sfdisk.c
+++ b/fdisk/sfdisk.c
@@ -2795,14 +2795,17 @@ do_pt_geom (char *dev, int silent) {
 /* for compatibility with earlier fdisk: provide option -s */
 static void
 do_size (char *dev, int silent) {
-    int fd;
+    int fd, rc;
     unsigned long long size;
 
     fd = my_open(dev, 0, silent);
     if (fd < 0)
 	return;
 
-    if (blkdev_get_sectors(fd, &size) == -1) {
+    rc = blkdev_get_sectors(fd, &size);
+    close(fd);
+
+    if (rc == -1) {
 	if (!silent) {
 	    perror(dev);
 	    fatal(_("Cannot get size of %s\n"), dev);
@@ -2812,18 +2815,12 @@ do_size (char *dev, int silent) {
 
     size /= 2;			/* convert sectors to blocks */
 
-    /* a CDROM drive without mounted CD yields MAXINT */
-    if (silent && size == ((1<<30)-1))
-      return;
-
     if (silent)
       printf("%s: %9llu\n", dev, size);
     else
       printf("%llu\n", size);
 
     total_size += size;
-
-    close(fd);
 }
 
 /*
-- 
2.7.4

